### DISCUSSION TITLE
How to add global styles w/o affecting storybook correctly in Nx Angular project

### DISCUSSION THREAD

**Discussion by DanK1368** [storybookjs/storybook#25647] [Help]
### Summary

We have an angular nx workspace project. We run storybook as a separate app. 
In our project.json, we currently include a bunch of styles files. They are added globally, which also means it overrides the default styles of storybook. They should be imported in a way so that only our components are affected. 
```json
    "build-storybook": {
     ...
        "styles": [
          "libs/shared/assets/styles/reset.css",
          "libs/shared/assets/styles/styles.scss",
          "libs/shared/assets/styles/theme.scss",
          "apps/space/src/styles.scss",
          "apps/portal/src/styles.scss",
          "libs/portal/assets/styles/portal-variables.scss",
        ],
        "stylePreprocessorOptions": {
          "includePaths": ["libs/shared/assets/styles", "libs/portal/assets/styles"]
        }
```

Should these be imported somehow in the `preview.ts` file ?

**Comment by Marklb** [comment]
I don't know if there is an ideal solution. If you don't use any inline components then the global styles could probably be excluded, outside of the iframe, but by default the docs use inline rendering that puts the components in the same document as the docs page itself.

As long as you are fine breaking any css rules that can't be preceded by another rule then you could create a global scss file that imports your styles in a rule that acts as a scope. The problem is that rules, such as `html {}` or `body {}` are not going to work, since the styles have to be at the root of the page to affect them.

Example:

```scss
// sb-styles.scss

// NOTE: You could use something else, such as a class, but I do a property that
// is set to the active theme.
[my-app-theme="base"] {
	@import 'libs/shared/assets/styles/reset.css';
	@import 'libs/shared/assets/styles/styles.scss';
	@import 'libs/shared/assets/styles/theme.scss';
	@import 'apps/space/src/styles.scss';
	@import 'apps/portal/src/styles.scss';
	@import 'libs/portal/assets/styles/portal-variables.scss';
}
```

Then you could use some sort of decorator in `.storybook/preview.ts` that wraps the story component in an element matching your prefixed rule. There are multiple ways to do that, but a simple way to do it for Angular is the following:

```ts
// .storybook/preview.ts
...
export decorators = [
  componentWrapperDecorator(story => `<span [attr.my-app-theme]="base">${story}</span>`)
]
```

**Comment by DanK1368** [comment]
@Marklb Thanks for getting back. 
So within the `.storybook` folder i created a `global-styles.scss` file.. and put these in there 
```typescript
[my-app-theme="base"] {
	@import 'libs/shared/assets/styles/reset.css';
	@import 'libs/shared/assets/styles/styles.scss';
	@import 'libs/shared/assets/styles/theme.scss';
	@import 'apps/space/src/styles.scss';
	@import 'apps/portal/src/styles.scss';
	@import 'libs/portal/assets/styles/portal-variables.scss';
}
```
also added the `componentWrapperDecorator` but im getting a warning that `[attr.my-app-theme]` is an `Unrecognized HTML Attribute` .. I also tried it with a class instead, but also doesn't work and none of the styles actually get applied.
